Zotero/Scenes/Detail/Annotation Popover/Views/AnnotationViewController.swift

```diff
@@ -213,8 +213,9 @@ final class AnnotationViewController: UIViewController {
             commentView.textObservable
                        .debounce(.milliseconds(500), scheduler: MainScheduler.instance)
                        .subscribe(with: self, onNext: { `self`, data in
-                           self.viewModel.process(action: .setComment(key: annotation.key, comment: data.0))
-                           if data.1 {
+                           guard let (text, needsHeightReload) = data else { return }
```

Comment:
Since you're changing this, can you switch to the style we agreed on previously? In this case remove `with: self` and just to `[weak self]` and `let self` in `guard` so that we don't need explicit `self`s in closure.

Anyway what was the leak here? The fact that we used `data` directly?

Reply:
Check https://github.com/zotero/zotero-ios/pull/831/commits/66585bb108b54a05de0ecf27945dfbae633d64c7

The leak is in the `textObservable`, because it creates a circular capture to it's owner. The change here just reflects that `data is now an optional tuple, and I just made it similar to the other place it is used, in `AnnotationView`.